masquerade.go

```diff
@@ -24,11 +23,19 @@ type Masquerade struct {
 	// IpAddress: pre-resolved ip address to use instead of Domain (if
 	// available)
 	IpAddress string
+}
+
+type masquerade struct {
```

Comment:
The crux of the fix is that we copy the provided Masquerade into an internal data structure, so whoever supplied us masquerades can do what they want with them without having to worry about synchronizing.

Reply:
And simply store and pass the object by value to reduce race condition